src/grasp_scorer.cpp

```diff
@@ -250,4 +250,120 @@ Eigen::Vector3d GraspScorer::scoreRotationsFromDesired(const Eigen::Isometry3d&
+    if (voxel_bb_max.x() < box_bb_min.x() || voxel_bb_min.x() > box_bb_max.x() || voxel_bb_max.y() < box_bb_min.y() ||
+        voxel_bb_min.y() > box_bb_max.y())
+    {
+      overlap_vector[voxel_id] = 0;
+    }
+    else
+    {
+      double x_overlap = std::max(voxel_bb_min.x(), box_bb_min.x()) - std::min(voxel_bb_max.x(), box_bb_max.x());
+      double y_overlap = std::max(voxel_bb_min.y(), box_bb_min.y()) - std::min(voxel_bb_max.y(), box_bb_max.y());
+      double overlap_fraction = (x_overlap * y_overlap) / voxel_area;
+      overlap_vector[voxel_id] = overlap_fraction;
+    }
+    // clang-format off
+    ROS_DEBUG_STREAM_NAMED("grasp_scorer.voxels", "box_bb_max   " << box_bb_max.x() << ",\t" << box_bb_max.y() << ",\t" << box_bb_max.z());
+    ROS_DEBUG_STREAM_NAMED("grasp_scorer.voxels", "box_bb_min   " << box_bb_min.x() << ",\t" << box_bb_min.y() << ",\t" << box_bb_min.z());
+    ROS_DEBUG_STREAM_NAMED("grasp_scorer.voxels", "voxel_bb_min " << voxel_bb_min.x() << ",\t" << voxel_bb_min.y() << ",\t" << voxel_bb_min.z());
+    ROS_DEBUG_STREAM_NAMED("grasp_scorer.voxels", "voxel_bb_max " << voxel_bb_max.x() << ",\t" << voxel_bb_max.y() << ",\t" << voxel_bb_max.z());
+    ROS_DEBUG_STREAM_NAMED("grasp_scorer.voxels", "overlap_score[" << voxel_id << "] = " << overlap_vector[voxel_id]);
+    // clang-format on
+
+    if (visual_tools)
+    {
+      Eigen::Isometry3d vbb1 =
+          object_pose * Eigen::Isometry3d(Eigen::Translation3d(voxel_bb_min) * Eigen::Quaterniond(1, 0, 0, 0));
+      Eigen::Isometry3d vbb2 =
+          object_pose * Eigen::Isometry3d(Eigen::Translation3d(voxel_bb_max) * Eigen::Quaterniond(1, 0, 0, 0));
+      Eigen::Isometry3d voxel_center_point =
+          grasp_pose_tcp *
+          Eigen::Isometry3d(Eigen::Translation3d(voxel->center_point_) * Eigen::Quaterniond(1, 0, 0, 0));
+
+      visual_tools->publishAxisLabeled(vbb1, "vbb1", rviz_visual_tools::SMALL);
+      visual_tools->publishAxisLabeled(vbb2, "vbb2", rviz_visual_tools::SMALL);
+      visual_tools->publishAxisLabeled(voxel_center_point, "voxel_center_point", rviz_visual_tools::SMALL);
+
+      double mesh_x_wid = voxel->x_width_ - 0.01;
+      double mesh_y_wid = voxel->y_width_ - 0.01;
+
+      if (overlap_vector[voxel_id] > 0.75)
+      {
+        visual_tools->publishWireframeCuboid(voxel_center_point, mesh_x_wid, mesh_y_wid, 0.001,
+                                             rviz_visual_tools::GREEN);
+      }
+      else if (overlap_vector[voxel_id] > 0.25)
+      {
+        visual_tools->publishWireframeCuboid(voxel_center_point, mesh_x_wid, mesh_y_wid, 0.001,
+                                             rviz_visual_tools::YELLOW);
+      }
+      else
+      {
+        visual_tools->publishWireframeCuboid(voxel_center_point, mesh_x_wid, mesh_y_wid, 0.001, rviz_visual_tools::RED);
+      }
+    }
+  }
+
+  double overhang_score = 0;
+  for (double voxel_overlap : overlap_vector)
+    overhang_score += voxel_overlap * voxel_overlap;
```

Comment:
Aren't we maximizing the score? This way we would maximize the overlap or am I wrong here?

Reply:
The larger the grasp score the better the candidate